rudder-core/src/main/scala/com/normation/rudder/hooks/RunHooks.scala

```diff
@@ -111,6 +111,35 @@ object HooksLogger extends Logger {
   override protected def _logger = LoggerFactory.getLogger("hooks")
 }
 
+sealed trait HookReturnCode
+object HookReturnCode {
+  sealed trait Continue extends HookReturnCode
+  sealed trait Stop     extends HookReturnCode {
+    def msg: String
+  }
+
+  //special return code
+  final case object Success extends Continue
+  final case class  Warning(code: Int, msg: String) extends Continue
+  final case class  Error(code: Int, msg: String) extends Stop
+  final case class  SystemError(msg: String) extends Stop
+
+  //special return code 100: it is a stop state, but in some case can lead to
+  //an user message that is tailored to explain that it is not a hook error)
+  final case class  Interrupt(msg: String) extends Stop
+}
```

Comment:
When used in type, these name are not really good (i.e: they are ok for the return code logic, but strange for the api user). 
I would propose using: 
- s/Continue/Success/
- s/Stop/Error/
- s/Success/OK/ 
- s/Error/ScriptError

So that we have a meaningfull Success -> OK and Success -> Warning and Error -> Interrupt, and Error -> ScriptError / Error -> SystemError keep a nice symetry. 

And so, type checking for an error in hook return code will have HookReturnCode.Error, which is meaningfull contrary to HookReturnCode.Stop.